### PR TITLE
Add SpongeExtension

### PR DESCRIPTION
**Pull request by windy1**
Signed-off-by: Walker Crouse <walkercrouse@hotmail.com>

### CODE REVIEW

File: src/main/java/org/spongepowered/plugin/meta/SpongeExtension.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.spongepowered.plugin.meta;
+
+public final class SpongeExtension {
+
+    public String assets; // public for mixin compatibility
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/plugin/meta/SpongeExtension.java]: What kind of "mixin compatibility"?

File: src/main/java/org/spongepowered/plugin/meta/SpongeExtension.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    public SpongeExtension(String assets) {
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/plugin/meta/SpongeExtension.java]: I think you can just use a no-arg constructor here, I'd only use a constructor for required things

Comment (windy1) [inline src/main/java/org/spongepowered/plugin/meta/SpongeExtension.java]: It should never be null according to ModMetadata in Forge and there are instances where you might want to initialize it right away.

Comment (Minecrell) [inline src/main/java/org/spongepowered/plugin/meta/SpongeExtension.java]: `PluginMetadata` is using null so unnecessary entries are not created in the JSON output (keep in mind Forge never serializes, it only deserializes). The no-arg constructor is just for consistence, don't want to add 5 new constructors to `PluginMetadata`

**Pull request merged**